File: ifif/python/G5_20437.py
import sys

sys.stdin = open("input.txt", "r")
T = int(input())
def solution(d, k):
    if not d:
        return [-1]
    min_num = 1e9
    max_num = 0
    for val in d.values():
        for i in range(len(val)-k+1):
            temp = val[i+k-1] - val[i] + 1
            min_num = min(min_num, temp)
            max_num = max(max_num, temp)
    return min_num, max_num


for _ in range(T):
    W = input()
    K = int(input())
    dic = dict()
    idx = -1
    for w in W:
        idx += 1
        dic.setdefault(w, [])
        dic[w].append(idx)
    for key, value in list(dic.items()):
        if len(value) < K:
            dic.pop(key)
    print(*solution(dic, K))

File: ifif/java/G5_20437.java
// 문자열 게임2
// 문자열, 슬라이딩 윈도우

import java.io.*;
import java.util.*;

public class G5_20437 {
    public static void main(String[] args) throws IOException {
        System.setIn(new FileInputStream("input.txt"));
        BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(sc.readLine());
        for (int t=0; t<T; t++) {
            String W = sc.readLine();
            int K = Integer.parseInt(sc.readLine());
            int[] answer = solution(W, K);
            for (int an : answer) {
                System.out.print(an);
                System.out.print(" ");
            }
            System.out.println();
        }
    }

    public static long countChar(String str, char ch) {
        return str.chars()
                .filter(c -> c == ch)
                .count();
    }
    private static int[] solution(String W, int K) {
        Map<Character, List<Integer>> map = new HashMap<>();
        for (int i=0; i<W.length(); i++) {
            if (countChar(W, W.charAt(i)) < K) {
                continue;
            }
            List<Integer> arrayList = map.getOrDefault(W.charAt(i), new ArrayList<>());
            arrayList.add(i);
            map.put(W.charAt(i), arrayList);
        }

        if (map.size() == 0) {
            return new int[]{-1};
        }
        int[] answer = new int[] {(int) 1e9, 0};
        for (List<Integer> val : map.values()) {
            for (int j=0; j<val.size()-K+1; j++) {
                int temp = val.get(j+K-1) - val.get(j) + 1;
                answer[0] = Math.min(answer[0], temp);
                answer[1] = Math.max(answer[1], temp);
            }
        }
        return answer;
    }
}
